### PR TITLE
Adds tab completion functionality as well as fixes

### PR DESCRIPTION
**Pull request by logan82902**

### CODE REVIEW

File: app/src/main/java/app/src/main/java/CliGameCompleter.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        // Iterate over your commands and find partial matches.
+        for (String command : new String[]{
+            "/newpuzzle", "/basepuzzle", "/guess", "/viewpuzzle", "/foundwords",
+            "/shuffle", "/cleansave", "/advancedsave", "/loadpuzzle", "/observestatus", "/help", "/exit"
+        }) {
+            if (command.startsWith(word)) {
+                matchingCommands.add(command);
+            }
+        }
+
```
Comment (island248) [inline app/src/main/java/app/src/main/java/CliGameCompleter.java:28]: Get rid of extra space please, thank you!  Should only have 1 blank line

File: app/src/main/java/app/src/main/java/CliGameCompleter.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        
+
+        // Add matching candidates.
+        for (String matchingCommand : matchingCommands) {
+            candidates.add(new Candidate(AttributedString.stripAnsi(matchingCommand), matchingCommand, null, null, null, null, true));
+        }
+
```
Comment (island248) [inline app/src/main/java/app/src/main/java/CliGameCompleter.java:32]: Get rid of empty line.

File: app/src/main/java/app/src/main/java/CliGameController.java [hunk 3]
```diff
@@ -2,114 +2,128 @@
 
 import java.io.FileNotFoundException;
 import java.util.Scanner;
-import java.io.IOException; 
+import java.io.IOException;
+import org.jline.reader.*;
+import org.jline.terminal.Terminal;
+import org.jline.terminal.TerminalBuilder;
+
```
Comment (island248) [inline app/src/main/java/app/src/main/java/CliGameController.java:10]: Get rid of extra blank line.

File: app/src/main/java/app/src/main/java/CliGameController.java [hunk 4]
```diff
@@ -2,114 +2,128 @@
...
+            lineReader = LineReaderBuilder.builder()
+                    .terminal(terminal)
+                    .completer(new CliGameCompleter()) 
+                    .build();
+
+            while (true) {
+                
+                System.out.print("\u001B[33m" + ">" + "\u001B[0m");
+                String input = lineReader.readLine();
+
```
Comment (island248) [inline app/src/main/java/app/src/main/java/CliGameController.java:38]: Only have 1 blank line

File: app/src/main/java/app/src/main/java/CliGameController.java [hunk 5]
```diff
@@ -2,114 +2,128 @@
...
+            while (true) {
+                
+                System.out.print("\u001B[33m" + ">" + "\u001B[0m");
+                String input = lineReader.readLine();
+                
+                
+                input = input.trim();
+                processInput(input);
+
```
Comment (island248) [inline app/src/main/java/app/src/main/java/CliGameController.java:41]: Get rid of blank line

**Review by island248 — CHANGES_REQUESTED** [review]
The functionality looks good, I am asking for some formatting to be changed.  Thank you!

**Review by logan82902 — COMMENTED** [review]
I have made the appropriate changes. Please merge when ready.

**Review by island248 — APPROVED** [review]

**Pull request merged**